Assets/Scripts/Game/UserInterfaceWindows/DaggerfallSpellMakerWindow.cs

```diff
@@ -743,6 +743,21 @@ private void BuyButton_OnMouseClick(BaseScreenComponent sender, Vector2 position
             //   Implement costs and remove gold from player and block if user does not have enough gold (message 1702)
             //   Spells will be free (both gold and spell points) during build-out of effect system.
 
+            var moneyAvailable = GameManager.Instance.PlayerEntity.GetGoldAmount();
+
+            if (moneyAvailable < totalGoldCost)
+            {
+                DaggerfallMessageBox mbNoMoney = DaggerfallUI.MessageBox(notEnoughGold);
+                mbNoMoney.ClickAnywhereToClose = true;
+                mbNoMoney.PreviousWindow = this;
+                mbNoMoney.Show();
+                return;
+            }
+            else
+            {
+                GameManager.Instance.PlayerEntity.DeductGoldAmount(totalGoldCost);
+            }
+
             // NOTES:
             //  In classic, player must have a spellbook item in their inventory (message 1703).
             //  In Daggerfall Unity, the spellbook is a permanent item that all players own and cannot be removed.
```

Comment:
I know this is unrelated to this patch, but I just realized that this comment is no longer true, and that the presence of a spellbook will need to be checked.
And when added, it will have to be checked before paying for the spell

Reply:
Actually what is relevant to your patch is that they're other conditions that are checked and that could prevent the spell from being made (no effects, no spell name), so paying should happen after all those tests

Reply:
Good call, thanks for your help on this.